File: Backend/models/category.model.js
import mongoose from 'mongoose'

const CategorySchema = new mongoose.Schema({
  product: { type: String, required: true },
  gender: { type: String, required: true }
})

CategorySchema.index({ product: 1, gender: 1 }, { unique: true })

export default mongoose.model('Category', CategorySchema)

File: Backend-Python/app/models/category_model.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.orm import relationship
from app.database.connection import Base


class Category(Base):
    __tablename__ = "categories"

    id = Column(Integer, primary_key=True, index=True)
    product = Column(String, nullable=False)
    gender = Column(String, nullable=False)
    products = relationship("Product", back_populates="category")
